crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -176,7 +177,27 @@ async fn download_block(
     let result = sequencer.block_by_number(block_number.into()).await;
 
     match result {
-        Ok(block) => Ok(DownloadBlock::Block(Box::new(block))),
+        Ok(block) => {
+            // Check if block hash is correct.
+            //
+            // NOTE: Blocks _after_ the 0.8.0 release and before the 0.8.2 release
+            // had their block hash calculated with an unknown sequencer address and thus don't match our computed
+            // value... Also, we've seen some sequencer bugs where the API returned a bogus sequencer_address
+            // for old blocks that had their block hashes computed with zero as the sequencer address.
+            let computed_block_hash = compute_block_hash(&block)?;
+            if let Some(block_hash) = block.block_hash {
+                if block_hash != computed_block_hash {
+                    let block_number = block.block_number.unwrap();
+                    tracing::warn!(
+                        ?block_number,
+                        ?block_hash,
+                        ?computed_block_hash,
+                        "Block hash mismatch"
+                    );
+                }
+            }
```

Comment:
We should only ship this once we have the block ranges we should be ignoring in place. That way we don't spam the logs on every block for no reason.

And then second; I would enforce this for all other blocks since failure means database corruption. That said, I am okay with making this a soft enforce initially (`error!` but don't fail the block) if we are uncertain about the correctness of our impl.

Reply:
Exactly, that's why it's still only a `warn!`. Once we have reliable information on the sequencer addresses for the different ranges I think we could actually start enforcing this.

Reply:
We should still mute it for the known broken ranges.

Reply:
Agreed. Wrote a program to find those ranges out but got stuck with a lot of block whose hashes doesn't match and couldn't yet figure out why.

Reply:
Probably using the other 3-4 possible sequencer addresses that were mentioned?

Reply:
Don't know yet. I've already asked Starkware for help.

Reply:
Turns out they don't know more potential addresses, just the single one we already have...

Reply:
This leads to us being unable to verify a range of testnet blocks, and the code now handles this known range.